crates/vfio-ioctls/src/fam.rs

```diff
@@ -33,7 +40,50 @@ fn vec_with_size_in_bytes<T: Default>(size_in_bytes: usize) -> Vec<T> {
 /// as a `Foo`. The remaining memory in the `Vec<Foo>` is for `entries`, which must be contiguous
 /// with `Foo`. This function is used to make the `Vec<Foo>` with enough space for `count` entries.
 pub(crate) fn vec_with_array_field<T: Default, F>(count: usize) -> Vec<T> {
-    let element_space = count * size_of::<F>();
-    let vec_size_bytes = size_of::<T>() + element_space;
+    let element_space = match count.checked_mul(size_of::<F>()) {
```

Comment:
Why is this still needed? I noticed you're using the fam struct from vmm-sys-util as well.

Reply:
It may be replaced by FamStruct, I hope do that with a separate PR:)

Reply:
I have double checked the code, and it seems FamStruct doesn't help much here. The structure used with `vec_with_array_field` have conflicts with FamStruct's assumption.
1) those structs has field "argsz" (in bytes) instead of `n_elements`
2) the following elements may be different type/size.

Need more thinking about whether it worth to switching to FamStruct.